### PR TITLE
IGNITE-23226 Services: deployment failed but service descriptor stay

### PR DESCRIPTION
**Pull request by vladnovoren**
…ket.

Thank you for submitting the pull request to the Apache Ignite.

In order to streamline the review of the contribution 
we ask you to ensure the following steps have been taken:

### The Contribution Checklist
- [ ] There is a single JIRA ticket related to the pull request. 
- [ ] The web-link to the pull request is attached to the JIRA ticket.
- [ ] The JIRA ticket has the _Patch Available_ state.
- [ ] The pull request body describes changes that have been made. 
The description explains _WHAT_ and _WHY_ was made instead of _HOW_.
- [ ] The pull request title is treated as the final commit message. 
The following pattern must be used: `IGNITE-XXXX Change summary` where `XXXX` - number of JIRA issue.
- [ ] A reviewer has been mentioned through the JIRA comments 
(see [the Maintainers list](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute#HowtoContribute-ReviewProcessandMaintainers)) 
- [ ] The pull request has been checked by the Teamcity Bot and 
the `green visa` attached to the JIRA ticket (see [TC.Bot: Check PR](https://mtcga.gridgain.com/prs.html))

### Notes
- [How to Contribute](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute)
- [Coding abbreviation rules](https://cwiki.apache.org/confluence/display/IGNITE/Abbreviation+Rules)
- [Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines)
- [Apache Ignite Teamcity Bot](https://cwiki.apache.org/confluence/display/IGNITE/Apache+Ignite+Teamcity+Bot)

If you need any help, please email dev@ignite.apache.org or ask anу advice on http://asf.slack.com _#ignite_ channel.

### CODE REVIEW

File: modules/core/src/test/java/org/apache/ignite/internal/processors/service/IgniteServiceDeploymentFailureTest.java [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+
+    /**
+     * Tests that service descriptors are clear after an attempt of deploying
+     * a service which throws an exception during initialization
+     *
+     * @throws Exception If failed.
+     * */
+    @Test
+    public void testServerDescriptorsOfFailedServices() throws Exception {
+        final int NOOP_SRVC_TOTAL_CNT = 20;
```
Comment (nizhikov) [inline modules/core/src/test/java/org/apache/ignite/internal/processors/service/IgniteServiceDeploymentFailureTest.java:146]: variables must be in camelCase -> noopSrvcTotalCnt here and below.

File: modules/core/src/test/java/org/apache/ignite/internal/processors/service/IgniteServiceDeploymentFailureTest.java [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+        /** {@inheritDoc} */
+        @Override public void init() throws Exception {
+            initCounter.incrementAndGet();
+            throw new Exception("Service init exception");
+        }
+    }
+
+    /**
+     * NoopService for testing.
+     * */
```
Comment (nizhikov) [inline modules/core/src/test/java/org/apache/ignite/internal/processors/service/IgniteServiceDeploymentFailureTest.java:127]: Looks like javadoc formatting wrong. Please, fix it.

File: modules/core/src/test/java/org/apache/ignite/internal/processors/service/IgniteServiceDeploymentFailureTest.java [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+            initCounter.incrementAndGet();
+        }
+    }
+
+    /**
+     * Tests that service descriptors are clear after an attempt of deploying
+     * a service which throws an exception during initialization
+     *
+     * @throws Exception If failed.
+     * */
```
Comment (nizhikov) [inline modules/core/src/test/java/org/apache/ignite/internal/processors/service/IgniteServiceDeploymentFailureTest.java:143]: Looks like javadoc formatting wrong. Please, fix it.

File: modules/core/src/test/java/org/apache/ignite/internal/processors/service/IgniteServiceDeploymentFailureTest.java [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+                        .setName(InitThrowingService.class.getSimpleName())
+                        .setService(new InitThrowingService())
+                        .setTotalCount(throwSrvcTotalCnt)
+                        .setMaxPerNodeCount(throwSrvcMaxPerNodeCnt)),
+                ServiceDeploymentException.class);
+
+        // Wait until the descriptors are updated on all nodes and check that there are no descriptors of an undeployed service.
+        assertTrue(waitForCondition(() -> {
+            return Ignition.allGrids().stream().allMatch(
+                    server -> server.services().serviceDescriptors().stream().noneMatch(
```
Comment (nizhikov) [inline modules/core/src/test/java/org/apache/ignite/internal/processors/service/IgniteServiceDeploymentFailureTest.java:183]: server -> node or ign

File: modules/core/src/test/java/org/apache/ignite/internal/processors/service/IgniteServiceDeploymentFailureTest.java (hunk 3 above)
Comment (vladnovoren) [reply modules/core/src/test/java/org/apache/ignite/internal/processors/service/IgniteServiceDeploymentFailureTest.java]: fixed

File: modules/core/src/test/java/org/apache/ignite/internal/processors/service/IgniteServiceDeploymentFailureTest.java (hunk 2 above)
Comment (vladnovoren) [reply modules/core/src/test/java/org/apache/ignite/internal/processors/service/IgniteServiceDeploymentFailureTest.java]: fixed

File: modules/core/src/test/java/org/apache/ignite/internal/processors/service/IgniteServiceDeploymentFailureTest.java (hunk 1 above)
Comment (vladnovoren) [reply modules/core/src/test/java/org/apache/ignite/internal/processors/service/IgniteServiceDeploymentFailureTest.java]: fixed

File: modules/core/src/main/java/org/apache/ignite/internal/processors/service/IgniteServiceProcessor.java [hunk 5]
```diff
@@ -1578,6 +1578,15 @@ void updateServicesTopologies(@NotNull final Map<IgniteUuid, Map<UUID, Integer>>
 
         try {
             updateServicesMap(deployedServices, fullTops);
+            for (Map.Entry<IgniteUuid, Map<UUID, Integer>> e : fullTops.entrySet()) {
```
Comment (nizhikov) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/service/IgniteServiceProcessor.java:1581]: Empty line after `updateServicesMap`.

File: modules/core/src/main/java/org/apache/ignite/internal/processors/service/IgniteServiceProcessor.java [hunk 6]
```diff
@@ -1578,6 +1578,15 @@ void updateServicesTopologies(@NotNull final Map<IgniteUuid, Map<UUID, Integer>>
...
+                // Checking if there are successful deployments.
+                // If none, service not deployed and must be removed from descriptors.
+                if (e.getValue().entrySet().stream().allMatch(nodeTop -> nodeTop.getValue() == 0)) {
+                    ServiceInfo desc = registeredServices.remove(e.getKey());
+                    if (desc != null)
```
Comment (nizhikov) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/service/IgniteServiceProcessor.java:1586]: Empty line after `ServiceInfo desc ...`

File: modules/core/src/test/java/org/apache/ignite/internal/processors/service/IgniteServiceDeploymentFailureTest.java [hunk 7]
```diff
@@ -0,0 +1,210 @@
...
+                .setName("TestDeploymentService")
+                .setService(((Class<Service>) extClsLdr.loadClass(NOOP_SERVICE_CLS_NAME)).getDeclaredConstructor().newInstance())
+                .setNodeFilter(((Class<IgnitePredicate<ClusterNode>>) extClsLdr.loadClass(NODE_FILTER_CLS_NAME))
+                        .getConstructor(UUID.class)
+                        .newInstance(cli.configuration().getNodeId()))
+                .setTotalCount(1);
+
+        assertThrowsWithCause(() -> cli.services().deploy(svcCfg), ServiceDeploymentException.class);
+
+        assertTrue(cli.services().serviceDescriptors().isEmpty());
```
Comment (nizhikov) [inline modules/core/src/test/java/org/apache/ignite/internal/processors/service/IgniteServiceDeploymentFailureTest.java:109]: Let's add the same check as we use in test below - `waitForCondition` all node doesn't have failed service in descriptors.

File: modules/core/src/test/java/org/apache/ignite/internal/processors/service/IgniteServiceDeploymentFailureTest.java [hunk 8]
```diff
@@ -0,0 +1,210 @@
...
+
+        ServiceConfiguration noopSrvcCfg = new ServiceConfiguration()
+                .setName(NoopService.class.getSimpleName())
+                .setService(new NoopService())
+                .setTotalCount(noopSrvcTotalCnt)
+                .setMaxPerNodeCount(noopSrvcMaxPerNodeCnt0);
+
+        // Deploying NoopService - should be completed successfully.
+        client.services().deploy(noopSrvcCfg);
+        // Check that the expected number of instances has been deployed
```
Comment (nizhikov) [inline modules/core/src/test/java/org/apache/ignite/internal/processors/service/IgniteServiceDeploymentFailureTest.java:141]: empty line after `        client.services().deploy(noopSrvcCfg);`

File: modules/core/src/test/java/org/apache/ignite/internal/processors/service/IgniteServiceDeploymentFailureTest.java [hunk 9]
```diff
@@ -0,0 +1,210 @@
...
+        assertThrowsWithCause(() -> client.services().deploy(new ServiceConfiguration()
+                        .setName(InitThrowingService.class.getSimpleName())
+                        .setService(new InitThrowingService())
+                        .setTotalCount(throwSrvcTotalCnt)
+                        .setMaxPerNodeCount(throwSrvcMaxPerNodeCnt)),
+                ServiceDeploymentException.class);
+
+        // Wait until the descriptors are updated on all nodes and check that there are no descriptors of an undeployed service.
+        assertTrue(waitForCondition(() -> {
+            return Ignition.allGrids().stream().allMatch(
```
Comment (nizhikov) [inline modules/core/src/test/java/org/apache/ignite/internal/processors/service/IgniteServiceDeploymentFailureTest.java:155]: `return` can be omited.

File: modules/core/src/test/java/org/apache/ignite/internal/processors/service/IgniteServiceDeploymentFailureTest.java [hunk 10]
```diff
@@ -0,0 +1,210 @@
...
+                    node -> node.services().serviceDescriptors().stream().noneMatch(
+                            desc -> desc.name().equals(InitThrowingService.class.getSimpleName()))
+            );
+        }, TIMEOUT));
+
+        client.services().cancel(NoopService.class.getSimpleName());
+
+        // Deploy some additional NoopService instances.
+        noopSrvcCfg.setMaxPerNodeCount(noopSrvcMaxPerNodeCnt1);
+        client.services().deploy(noopSrvcCfg);
```
Comment (nizhikov) [inline modules/core/src/test/java/org/apache/ignite/internal/processors/service/IgniteServiceDeploymentFailureTest.java:165]: empty line after `noopSrvcCfg.set ...`

File: modules/core/src/test/java/org/apache/ignite/internal/processors/service/IgniteServiceDeploymentFailureTest.java [hunk 11]
```diff
@@ -0,0 +1,210 @@
...
+     * Service that throws an exception in init.
+     */
+    private static class InitThrowingService implements Service {
+        /** */
+        public static final AtomicInteger initCounter = new AtomicInteger();
+
+        /** {@inheritDoc} */
+        @Override public void init() throws Exception {
+            initCounter.incrementAndGet();
+            throw new Exception("Service init exception");
```
Comment (nizhikov) [inline modules/core/src/test/java/org/apache/ignite/internal/processors/service/IgniteServiceDeploymentFailureTest.java:194]: empty line after `initCounter...`

File: modules/core/src/test/java/org/apache/ignite/internal/processors/service/IgniteServiceDeploymentFailureTest.java (hunk 10 above)
Comment (vladnovoren) [reply modules/core/src/test/java/org/apache/ignite/internal/processors/service/IgniteServiceDeploymentFailureTest.java:165]: fixed

File: modules/core/src/test/java/org/apache/ignite/internal/processors/service/IgniteServiceDeploymentFailureTest.java (hunk 8 above)
Comment (vladnovoren) [reply modules/core/src/test/java/org/apache/ignite/internal/processors/service/IgniteServiceDeploymentFailureTest.java:142]: fixed

File: modules/core/src/main/java/org/apache/ignite/internal/processors/service/IgniteServiceProcessor.java (hunk 6 above)
Comment (vladnovoren) [reply modules/core/src/main/java/org/apache/ignite/internal/processors/service/IgniteServiceProcessor.java:1588]: fixed

File: modules/core/src/main/java/org/apache/ignite/internal/processors/service/IgniteServiceProcessor.java (hunk 5 above)
Comment (vladnovoren) [reply modules/core/src/main/java/org/apache/ignite/internal/processors/service/IgniteServiceProcessor.java:1582]: fixed

File: modules/core/src/test/java/org/apache/ignite/internal/processors/service/IgniteServiceDeploymentFailureTest.java (hunk 7 above)
Comment (vladnovoren) [reply modules/core/src/test/java/org/apache/ignite/internal/processors/service/IgniteServiceDeploymentFailureTest.java]: done

File: modules/core/src/test/java/org/apache/ignite/internal/processors/service/IgniteServiceDeploymentFailureTest.java [hunk 12]
```diff
@@ -0,0 +1,231 @@
...
+        // Deploy InitThrowingService that throws an exception in init - should not be deployed.
+        assertThrowsWithCause(() -> client.services().deploy(new ServiceConfiguration()
+                        .setName(InitThrowingService.class.getSimpleName())
+                        .setService(new InitThrowingService())
+                        .setTotalCount(throwSrvcTotalCnt)
+                        .setMaxPerNodeCount(throwSrvcMaxPerNodeCnt)),
+                ServiceDeploymentException.class);
+
+        // Wait until the descriptors are updated on all nodes and check that there are no descriptors of an undeployed service.
+        assertTrue(waitForCondition(() -> noDescriptorInClusterForService(InitThrowingService.class.getSimpleName()), TIMEOUT));
```
Comment (nizhikov) [inline modules/core/src/test/java/org/apache/ignite/internal/processors/service/IgniteServiceDeploymentFailureTest.java:159]: Let's add check that NoopService that was deployed successfully present in descriptors on all nodes.

File: modules/core/src/test/java/org/apache/ignite/internal/processors/service/IgniteServiceDeploymentFailureTest.java [hunk 13]
```diff
@@ -0,0 +1,231 @@
...
+        extClsLdr = null;
+    }
+
+    /**
+     * Tests that deploying a service with a missing class causes a ServiceDeploymentException.
+     *
+     * @throws Exception If failed.
+     */
+    @Test
+    public void testFailWhenClassNotFound() throws Exception {
```
Comment (nizhikov) [inline modules/core/src/test/java/org/apache/ignite/internal/processors/service/IgniteServiceDeploymentFailureTest.java:96]: Let's add similar test but when service deployed via IgniteConfiguration

File: modules/core/src/test/java/org/apache/ignite/internal/processors/service/IgniteServiceDeploymentFailureTest.java [hunk 14]
```diff
@@ -0,0 +1,231 @@
...
+    }
+
+    /**
+     * Tests that service descriptors are clear after an attempt of deploying
+     * a service which throws an exception during initialization
+     *
+     * @throws Exception If failed.
+     */
+    @Test
+    public void testServerDescriptorsOfFailedServices() throws Exception {
```
Comment (nizhikov) [inline modules/core/src/test/java/org/apache/ignite/internal/processors/service/IgniteServiceDeploymentFailureTest.java:122]: Let's add similar test but when service deployed statically via IgniteConfiguration

File: modules/core/src/test/java/org/apache/ignite/internal/processors/service/IgniteServiceDeploymentFailureTest.java [hunk 15]
```diff
@@ -0,0 +1,267 @@
...
+
+    /**
+     * Tests that service descriptors are clear after an attempt of deploying a service which throws an exception
+     * during initialization. Dynamic configuration for services is used.
+     *
+     * @throws Exception If failed.
+     */
+    @Test
+    public void testFailedServiceDescriptorsDynamicConfiguration() throws Exception {
+        final int noopSrvcTotalCnt = 20;
```
Comment (nizhikov) [inline modules/core/src/test/java/org/apache/ignite/internal/processors/service/IgniteServiceDeploymentFailureTest.java:122]: noopSrvcTotalCnt = SERVER_NODES_CNT * noopSrvcMaxPerNodeCnt1;

Comment (nizhikov) [reply modules/core/src/test/java/org/apache/ignite/internal/processors/service/IgniteServiceDeploymentFailureTest.java:122]: Let's inline this variable

File: modules/core/src/test/java/org/apache/ignite/internal/processors/service/IgniteServiceDeploymentFailureTest.java [hunk 16]
```diff
@@ -0,0 +1,267 @@
...
+
+        final int noopSrvcMaxPerNodeCnt0 = 2;
+        final int noopSrvcTotalCnt0 = noopSrvcMaxPerNodeCnt0 * SERVER_NODES_CNT;
+
+        final int noopSrvcMaxPerNodeCnt1 = 4;
+
+        final int throwSrvcTotalCnt = 10;
```
Comment (nizhikov) [inline modules/core/src/test/java/org/apache/ignite/internal/processors/service/IgniteServiceDeploymentFailureTest.java:129]: Single usage, can be inlined.

File: modules/core/src/test/java/org/apache/ignite/internal/processors/service/IgniteServiceDeploymentFailureTest.java [hunk 17]
```diff
@@ -0,0 +1,267 @@
...
+        final int throwSrvcMaxPerNodeCnt = 2;
```
Comment (nizhikov) [inline modules/core/src/test/java/org/apache/ignite/internal/processors/service/IgniteServiceDeploymentFailureTest.java:130]: Single usage, can be inlined

File: modules/core/src/test/java/org/apache/ignite/internal/processors/service/IgniteServiceDeploymentFailureTest.java (hunk 16 above)
Comment (nizhikov) [reply modules/core/src/test/java/org/apache/ignite/internal/processors/service/IgniteServiceDeploymentFailureTest.java:129]: SERVER_NODES_CNT * 2

File: modules/core/src/test/java/org/apache/ignite/internal/processors/service/IgniteServiceDeploymentFailureTest.java [hunk 18]
```diff
@@ -0,0 +1,267 @@
...
+    /**
+     * Tests that service descriptors are clear after an attempt of deploying a service which throws an exception
+     * during initialization. Static configuration for services is used.
+     *
+     * @throws Exception If failed.
+     */
+    @Test
+    public void testFailedServiceDescriptorsStaticConfiguration() throws Exception {
+        final int noopSrvcTotalCnt = 20;
+        final int initThrowingSrvcTotalCnt = 20;
```
Comment (nizhikov) [inline modules/core/src/test/java/org/apache/ignite/internal/processors/service/IgniteServiceDeploymentFailureTest.java:187]: Single usage, can be inlined

File: modules/core/src/test/java/org/apache/ignite/internal/processors/service/IgniteServiceDeploymentFailureTest.java [hunk 19]
```diff
@@ -0,0 +1,267 @@
...
+                .setService(new NoopService())
+                .setTotalCount(noopSrvcTotalCnt)
+                .setMaxPerNodeCount(noopSrvcMaxPerNodeCnt0);
+
+        // Deploying NoopService - should be completed successfully.
+        client.services().deploy(noopSrvcCfg);
+
+        // Check that the expected number of instances has been deployed
+        assertEquals(noopSrvcTotalCnt0,
+                totalInstancesCount(client, NoopService.class.getSimpleName()));
```
Comment (nizhikov) [inline modules/core/src/test/java/org/apache/ignite/internal/processors/service/IgniteServiceDeploymentFailureTest.java:150]:
```suggestion
        assertEquals(
            "Expected number of instances must be deployed",
            noopSrvcTotalCnt0,
            totalInstancesCount(client, NoopService.class.getSimpleName())
        );
```

File: modules/core/src/test/java/org/apache/ignite/internal/processors/service/IgniteServiceDeploymentFailureTest.java (hunk 15 above)
Comment (vladnovoren) [reply modules/core/src/test/java/org/apache/ignite/internal/processors/service/IgniteServiceDeploymentFailureTest.java:122]: It is used twice, why should we inline it?

File: modules/core/src/test/java/org/apache/ignite/internal/processors/service/IgniteServiceDeploymentFailureTest.java (hunk 19 above)
Comment (nizhikov) [inline modules/core/src/test/java/org/apache/ignite/internal/processors/service/IgniteServiceDeploymentFailureTest.java:150]:
```suggestion
        assertEquals(
            "Expected number of instances must be deployed",
            noopSrvcTotalCnt0,
            totalInstancesCount(client, NoopService.class.getSimpleName())
        );
```

File: modules/core/src/test/java/org/apache/ignite/internal/processors/service/IgniteServiceDeploymentFailureTest.java [hunk 20]
```diff
@@ -0,0 +1,267 @@
...
+        // Deploy InitThrowingService that throws an exception in init - should not be deployed.
+        assertThrowsWithCause(() -> client.services().deploy(new ServiceConfiguration()
+                        .setName(InitThrowingService.class.getSimpleName())
+                        .setService(new InitThrowingService())
+                        .setTotalCount(throwSrvcTotalCnt)
+                        .setMaxPerNodeCount(throwSrvcMaxPerNodeCnt)),
+                ServiceDeploymentException.class);
+
+        // Wait until the descriptors are updated on all nodes and check that there are no descriptors of an undeployed service.
+        assertTrue(waitForCondition(() -> noDescriptorInClusterForService(InitThrowingService.class.getSimpleName()), TIMEOUT));
```
Comment (nizhikov) [inline modules/core/src/test/java/org/apache/ignite/internal/processors/service/IgniteServiceDeploymentFailureTest.java:161]:
```suggestion
        assertTrue(
            "Descriptor of failed service must be removed from all nodes",
            waitForCondition(() -> noDescriptorInClusterForService(InitThrowingService.class.getSimpleName()), TIMEOUT)
        );
```

File: modules/core/src/test/java/org/apache/ignite/internal/processors/service/IgniteServiceDeploymentFailureTest.java [hunk 21]
```diff
@@ -0,0 +1,267 @@
...
+
+        // Check that NoopService that is deployed successfully previously is presented in descriptors.
+        assertEquals(noopSrvcTotalCnt0, totalInstancesCount(client, NoopService.class.getSimpleName()));
```
Comment (nizhikov) [inline modules/core/src/test/java/org/apache/ignite/internal/processors/service/IgniteServiceDeploymentFailureTest.java:164]:
```suggestion
        assertEquals(
            "Deployed service must be presented in cluster",
            noopSrvcMaxPerNodeCnt0 * SERVER_NODES_CNT, 
            totalInstancesCount(client, NoopService.class.getSimpleName())
        );
```

File: modules/core/src/test/java/org/apache/ignite/internal/processors/service/IgniteServiceDeploymentFailureTest.java [hunk 22]
```diff
@@ -0,0 +1,267 @@
...
+
+        client.services().cancel(NoopService.class.getSimpleName());
+
+        // Deploy some additional NoopService instances.
+        noopSrvcCfg.setMaxPerNodeCount(noopSrvcMaxPerNodeCnt1);
+
+        client.services().deploy(noopSrvcCfg);
+
+        // Check that the expected number of NoopService instances has been deployed.
+        assertEquals(noopSrvcTotalCnt, totalInstancesCount(client, NoopService.class.getSimpleName()));
```
Comment (nizhikov) [inline modules/core/src/test/java/org/apache/ignite/internal/processors/service/IgniteServiceDeploymentFailureTest.java:174]:
```suggestion
        assertEquals(
            "Deploy completed successfully",
            noopSrvcTotalCnt,
            totalInstancesCount(client, NoopService.class.getSimpleName())
        );
```

File: modules/core/src/test/java/org/apache/ignite/internal/processors/service/IgniteServiceDeploymentFailureTest.java (hunk 15 above)
Comment (nizhikov) [reply modules/core/src/test/java/org/apache/ignite/internal/processors/service/IgniteServiceDeploymentFailureTest.java:122]: Simple calculated value - no reason to create separate variable.

File: modules/core/src/test/java/org/apache/ignite/internal/processors/service/IgniteServiceDeploymentFailureTest.java [hunk 23]
```diff
@@ -0,0 +1,267 @@
...
+                .setName(NoopService.class.getSimpleName())
+                .setService(new NoopService())
+                .setTotalCount(noopSrvcTotalCnt);
+
+        ServiceConfiguration initThrowingCfg = new ServiceConfiguration()
+                .setName(InitThrowingService.class.getSimpleName())
+                .setService(new InitThrowingService())
+                .setTotalCount(initThrowingSrvcTotalCnt);
+
+        startGrids(SERVER_NODES_CNT - 1);
```
Comment (nizhikov) [inline modules/core/src/test/java/org/apache/ignite/internal/processors/service/IgniteServiceDeploymentFailureTest.java:199]: Let's start some client nodes here.

File: modules/core/src/test/java/org/apache/ignite/internal/processors/service/IgniteServiceDeploymentFailureTest.java [hunk 24]
```diff
@@ -0,0 +1,273 @@
...
+            .setService(new NoopService())
+            .setTotalCount(noopSrvcTotalCnt);
+
+        ServiceConfiguration initThrowingCfg = new ServiceConfiguration()
+            .setName(InitThrowingService.class.getSimpleName())
+            .setService(new InitThrowingService())
+            .setTotalCount(20);
+
+        startGrids(SERVER_NODES_CNT - 1);
+        for (int i = 0; i < CLIENT_NODES_CNT; i++)
```
Comment (nizhikov) [inline modules/core/src/test/java/org/apache/ignite/internal/processors/service/IgniteServiceDeploymentFailureTest.java:202]: empty line before this.

File: modules/core/src/test/java/org/apache/ignite/internal/processors/service/IgniteServiceDeploymentFailureTest.java [hunk 25]
```diff
@@ -0,0 +1,273 @@
...
+            startClientGrid(SERVER_NODES_CNT + i);
+        IgniteEx ign = startGrid(getConfiguration().setServiceConfiguration(noopCfg, initThrowingCfg));
```
Comment (nizhikov) [inline modules/core/src/test/java/org/apache/ignite/internal/processors/service/IgniteServiceDeploymentFailureTest.java:204]: empty line before this.

**Review by nizhikov — APPROVED** [review]

**Pull request merged**